smithy-openapi/src/main/java/software/amazon/smithy/openapi/fromsmithy/protocols/AbstractRestProtocol.java

```diff
@@ -526,12 +529,19 @@ private void createResponseDocumentIfNeeded(
         //
         // **NOTE: this same blurb applies to why we do this on input.**
         Schema schema = createDocumentSchema(context, operationOrError, bindings, messageType);
-        String synthesizedName = operationOrError.getId().getName() + "ResponseContent";
+        String synthesizedName = stripNonAlphaNumericCharsIfNecessary(context, operationOrError.getId().getName())
+                + "ResponseContent";
         String pointer = context.putSynthesizedSchema(synthesizedName, schema);
         MediaTypeObject mediaTypeObject = MediaTypeObject.builder()
                 .schema(Schema.builder().ref(pointer).build())
                 .build();
 
         responseBuilder.putContent(mediaType, mediaTypeObject);
     }
+
+    private String stripNonAlphaNumericCharsIfNecessary(Context<T> context, String name) {
+        return context.getConfig().getAlphanumericOnlyRefs()
```

Comment:
Gonna want to at least log an info / warning here saying what's happening

Reply:
Added logging in cd8a781.